### ISSUE TITLE
HtmlReporter SyntaxError

### ISSUE THREAD

**Issue by fkowal** [scalatest/scalatest#1579]
I am using scalatest 3.0.7 with scala 2.12.8
After running my tests when I open regenerated HtmlReport
The browser breaks with:
SyntaxError: expected expression, got '<' index.html:15

the generated html body contains this:
```
{ unparsedXml(
            "var tagMap = {};" + "\n" +     
            "var SUCCEEDED_BIT = 1;" + "\n" + 
            "var FAILED_BIT = 2;" + "\n" + 
....
```
https://github.com/scalatest/scalatest/blob/3.0.x/scalatest/src/main/scala/org/scalatest/tools/HtmlReporter.scala#L523

the reason for this is the scala.xml does not embed/execute the given block
inside `<![CDATA[`

This bug was introduced here:
https://github.com/scalatest/scalatest/commit/545917af623b9e2c388c4e9be2f57e48d87df062

As a solution I'd like to propose
Creating a new file `scalatest/src/main/resources/org/scalatest/HtmlReporter.js` contaning all the javascript functions embedded in the [HtmlReporter](https://github.com/scalatest/scalatest/blob/3.0.x/scalatest/src/main/scala/org/scalatest/tools/HtmlReporter.scala)

I would prepare such a PR, just need to make sure we're on the same page with the proposed solution

**Comment by fkowal** [comment]
Another perhaps a cleaner approach with a possible benefit of removing the dependency on scala.xml.
Is to completely move all the .html code into resources/ static files.
Serializing the data required by the .html into json, generating a .js file that the .html would load.

**Comment by cheeseng** [comment]
Hi @fkowal fyi we just merged a PR to fix this: 

https://github.com/scalatest/scalatest/pull/1596

I am back-porting the fix to 3.0.x to be included in 3.0.8 final.

Cheers.

**Comment by fkowal** [comment]
Great job. Thanks
